### ISSUE TITLE
Using Gatsby v2 and styled-components v4 together, extending component styles no longer work

### ISSUE THREAD

**Issue by gi-alec** [gatsbyjs/gatsby#8224]
Using Gatsby v2 and styled-components v4 together, extending component styles no longer work.
**There is a thread open on styled-components spectrum group [here](https://spectrum.chat/styled-components/?thread=4a2f04dd-b37a-47ea-8633-82f7dc8ba562).**

I noticed the elements in question show a class name (or 2) in the DOM, but the styles for that class are empty.

[Gist of the project structure](https://gist.github.com/gi-alec/392f7a0e2f87c7237fb69e120b8949e0)
[Gist of the offending code](https://gist.github.com/gi-alec/9d3db3ff65d4c00ddf9e258bd3d7af24).
[Gist of stylesheet comparisons](https://gist.github.com/gi-alec/16f62932a2bf40c40806cf34df830eae)

### Steps to reproduce
Create styled component
```javascript
const Button = styled.button`
border: 1px solid red;
`
```
Extend component
```javascript
const Button = styled(Button)`
border: 1px solid green;
`
```

### Expected result

Styles should be appending to the stylesheet output by styled-components

### Actual result
Classes for extended components are empty

### Environment

  System:
    OS: macOS High Sierra 10.13.4
    CPU: x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.6.0 - ~/.nvm/versions/node/v10.6.0/bin/node
    npm: 6.4.1 - ~/.nvm/versions/node/v10.6.0/bin/npm
  Browsers:
    Chrome: 69.0.3497.92
    Firefox: 61.0.2
    Safari: 11.1
  npmPackages:
    gatsby: ^2.0.0-rc.25 => 2.0.0-rc.25 
    gatsby-image: ^2.0.0-rc.2 => 2.0.0-rc.2 
    gatsby-plugin-canonical-urls: ^2.0.0-rc.1 => 2.0.0-rc.1 
    gatsby-plugin-feed: ^2.0.0-rc.2 => 2.0.0-rc.2 
    gatsby-plugin-google-analytics: ^2.0.0-rc.2 => 2.0.0-rc.2 
    gatsby-plugin-google-fonts: 0.0.4 => 0.0.4 
    gatsby-plugin-manifest: ^2.0.2-rc.1 => 2.0.2-rc.1 
    gatsby-plugin-netlify: ^2.0.0-rc.6 => 2.0.0-rc.6 
    gatsby-plugin-nprogress: ^2.0.0-rc.1 => 2.0.0-rc.1 
    gatsby-plugin-offline: ^2.0.0-rc.8 => 2.0.0-rc.8 
    gatsby-plugin-react-helmet: ^3.0.0-rc.1 => 3.0.0-rc.1 
    gatsby-plugin-sharp: ^2.0.0-rc.7 => 2.0.0-rc.7 
    gatsby-plugin-sitemap: ^2.0.0-rc.2 => 2.0.0-rc.2 
    gatsby-plugin-styled-components: ^3.0.0-rc.5 => 3.0.0-rc.5 
    gatsby-remark-images-contentful: ^2.0.0-rc.6 => 2.0.0-rc.6 
    gatsby-source-contentful: ^2.0.1-rc.9 => 2.0.1-rc.9 
    gatsby-source-filesystem: ^2.0.1-rc.6 => 2.0.1-rc.6 
    gatsby-transformer-remark: ^2.1.1-rc.5 => 2.1.1-rc.5 
    gatsby-transformer-sharp: ^2.1.1-rc.3 => 2.1.1-rc.3 
  npmGlobalPackages:
    gatsby-cli: 1.1.58

Edit: Here is a minimal repro
https://github.com/gi-alec/gatsby-v2-styled-components-v4

**Comment by gi-alec** [comment]
Side note. For some reason gatsby is throwing an error if gatsby-link isn't installed, despite being on version 2

**Comment by kakadiadarpan** [comment]
@probablyup solution provided by @gi-alec  in this comment https://github.com/gatsbyjs/gatsby/issues/8224#issuecomment-422157559 solves the issue. Can you try using that?

**Comment by probablyup** [comment]
So I made a PR to hopefully get around this: https://github.com/styled-components/styled-components/pull/2021

You can test it out with
```
npm install styled-components@4.0.0-beta.7-hoist
```

and just remove the resolutions package.json entry beforehand

**Comment by smakosh** [comment]
Use
```
<Button as={Button} />

instead of

const Button = styled(Button)`
border: 1px solid green;
`
```

**Comment by kakadiadarpan** [comment]
Fixed in styled-components [v4.0.0-beta.8](https://github.com/styled-components/styled-components/releases/tag/v4.0.0-beta.8)
